diff --git a/node_modules/eleventy-plugin-gen-favicons/favicon-gen.js b/node_modules/eleventy-plugin-gen-favicons/favicon-gen.js
index 10ef1f8..b97ce3f 100644
--- a/node_modules/eleventy-plugin-gen-favicons/favicon-gen.js
+++ b/node_modules/eleventy-plugin-gen-favicons/favicon-gen.js
@@ -30,7 +30,7 @@ const cacheByFile = {};
 
 // expects square img
 const resizedSharp = async (fileName, fileMeta, newDim) => {
-  const opts = fileMeta.density ? {density: (newDim / fileMeta.width) * fileMeta.density} : {};
+  const opts = fileMeta.density ? {density: Math.max(72,(newDim / fileMeta.width) * fileMeta.density)} : {};
   return sharp(fileName, opts).resize(newDim, newDim).png();
 };
 
@@ -72,7 +72,7 @@ module.exports = async (srcFile, outputDir, opts) => {
   const mtime = (await fsp.stat(srcFile)).mtime;
   const [cachedMtime, cachedOpts] = cacheByFile[`${srcFile}|${outputDir}`] || [0, {}];
 
-  const srcMetadata = await sharp(srcFile).metadata();
+  const srcMetadata = await sharp(srcFile, {failOn: 'none'}).png().metadata();
   const srcIsSvg = srcMetadata.format === 'svg';
 
   const {width, height} = srcMetadata;
